include/hc_printf.hpp

```diff
@@ -317,6 +317,7 @@ static inline void processPrintfPackets(PrintfPacket* packets, const unsigned in
       if (!std::regex_search(formatString, specifierMatches, specifierPattern)) {
         // More printf argument than format specifier??
         // Just skip to the next printf request
+        i++;
```

Comment:
you are only skipping one unused arg, it should really be

`i+=(numPrintfArgs - j);`